src/utils/find.ts

```diff
@@ -20,6 +20,10 @@ export function matches(
   // do not return none Vue components
   if (!node.component) return false
 
+  if (node.type === selector) {
+    return true
+  }
+
```

Comment:
Ain't that exactly what's done below, but maybe prevented by the condition?
I think it would be better to just extend the condition so it works with functional component (with some comments if possible)

Reply:
Good idea, I will revisit this logic. The entire function relies on a lot of side effects and mutation (mutating the  `vnodes` array). I think we should consider rewriting it in a more immutable manner potentially 🤔